lib/pmdtester/builders/pmd_report_builder.rb

```diff
@@ -0,0 +1,81 @@
+require 'fileutils'
+require '../cmd'
+module PmdTester
+  class PmdReportBuilder
+    def initialize(branch_config, projects, local_git_repo, pmd_branch_name, rule_sets)
+      @branch_config = branch_config
+      @projects = projects
+      @local_git_repo = local_git_repo
+      @pmd_branch_name = pmd_branch_name
+      @rule_sets = rule_sets
+      @pwd = Dir.getwd
+    end
+
+    def create_repositories_dir
+      @repositories_dir = @pwd + "/target/repositories"
+      FileUtils.mkdir(@repositories_dir) unless File.directory?(@repositories_dir)
+    end
+
+    def get_projects
+      puts "Cloning projects started"
+
+      create_repositories_dir
+
+      @projects.each do |project|
+        path = @repositories_dir + project.name
+        clone_cmd = project.type + " clone " + project.connection + " " + path
+        PmdTester::Cmd.execute(clone_cmd) unless File::exist?(path)
+        end
+    end
+
+    def get_pmd_binary_file
+      Dir.chdir(@local_git_repo)
+
+      checkout_cmd = "git checkout " + @pmd_branch_name
+      PmdTester::Cmd.execute(checkout_cmd)
+
+      package_cmd = "./mvnw clean package -Dpmd.test.skip=true -Dpmd.skip=true"
+      PmdTester::Cmd.execute(package_cmd)
+
+      version_cmd = "./mvnw -q -Dexec.executable=\"echo\" -Dexec.args='${project.version}' " +
+          "--non-recursive org.codehaus.mojo:exec-maven-plugin:1.5.0:exec | tail -1"
```

Comment:
The "tail -1" part is not needed anymore. It was a problem of an older maven wrapper version, but since then, we already updated.
(the old maven wrapper output first a line issueing the wrapper version itself before executing the command...)